scalafmt-tests/src/test/resources/dotty/MatchType.stat

```diff
@@ -0,0 +1,83 @@
+#
+<<< simple match type
+type Elem[X] = X match {
+             case String => Char
+             case Array[t] => t
+}
+>>>
+type Elem[X] = X match {
+  case String   => Char
+  case Array[t] => t
+}
+<<< type bound match type
+type Head[X <: Tuple] = X match {
+           case (x1, ?) => 
+           x1
+         }
+>>>
+type Head[X <: Tuple] = X match {
+  case (x1, ?) =>
+    x1
+}
+<<< recursive match type
+type Len[X] <: Int = X match {
+           case Unit => 0
+           case x *: xs => S[Len[xs]]}
+>>>
+type Len[X] <: Int = X match {
+  case Unit    => 0
+  case x *: xs => S[Len[xs]]
```

Comment:
side question: do we have a rule which formats this type of expression (*)?

Reply:
I think this gets translated as an Type.ApplyInfix so basically it should format anything according to existing settings. It seems to be checked in a couple of places.